llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -639,7 +641,9 @@ static Instruction *foldCtpop(IntrinsicInst &II, InstCombinerImpl &IC) {
   // ctpop (zext X) --> zext (ctpop X)
   if (match(Op0, m_OneUse(m_ZExt(m_Value(X))))) {
     Value *NarrowPop = IC.Builder.CreateUnaryIntrinsic(Intrinsic::ctpop, X);
-    return CastInst::Create(Instruction::ZExt, NarrowPop, Ty);
+    Instruction *Zext = CastInst::Create(Instruction::ZExt, NarrowPop, Ty);
+    Zext->setNonNeg();
```

Comment:
Is this correct for i1?

Reply:
It is incorrect for i1 and i2.
Alive2: https://alive2.llvm.org/ce/z/CvSF2z